python-checks/src/test/resources/checks/datetime_constructor_pytz_timezone.py

```diff
@@ -0,0 +1,69 @@
+    some_timezone = pytz.timezone('US/Eastern')
+                   #^^^^^^^^^^^^^^^^^^^^^^^^^^^> 1 {{The pytz.timezone is created here.}}
+    dt1 = datetime.datetime(2022, 1, 1, tzinfo=some_timezone)  # Noncompliant {{Don't pass a "pytz.timezone" to the "datetime.datetime" constructor.}}
+                                       #^^^^^^^^^^^^^^^^^^^^
+    return dt1
+
+some_timezone_1 = pytz.timezone('US/Eastern')
+dt1 = datetime.datetime(2022, 1, 1, tzinfo=some_timezone_1)  # FN because of ReachingDefinitionsAnalysis limitations : it only runs in functions
+
+def compliant():
+    other_timezone = datetime.timezone(datetime.timedelta(hours=2))
+    dt2 = datetime.datetime(2022, 1, 1, tzinfo=datetime.timezone(datetime.timedelta(hours=2)))
+    dt3 = datetime.datetime(2022, 1, 1, tzinfo=other_timezone)
+
+    dt4 = datetime.datetime(2022, 1, 1, tzinfo=None)
+
+    dt5 = datetime.datetime(2022, 1, 1)
+    def some_function():
+        return None
+    unrelated_call = some_function()
+    unknown_call()
+    unknown_call(True)
+    unknown_call(True, some_keyword=True)
+
+    dt6 =  datetime.datetime(2022, 1, 1, tzinfo=some_function())
+    dt7 =  datetime.datetime(2022, 1, 1, tzinfo=unrelated_call)
+
+    dt8 = datetime.datetime()
+def with_tuple_unpacking():
+    date_tuple = [2024, 2, 28]
+    some_timezone_2 = pytz.timezone('US/Eastern')
+    dt9 = datetime.datetime(*date_tuple, tzinfo=some_timezone_2)  # Noncompliant
+    dt9 = datetime.datetime(*date_tuple, tzinfo=pytz.timezone('US/Eastern'))  # Noncompliant
+
+def multiple_definitions():
+    if random():
+        some_timezone_3 = pytz.timezone('US/Eastern')
+                         #^^^^^^^^^^^^^^^^^^^^^^^^^^^> 1 {{The pytz.timezone is created here.}}
+    else:
+        some_timezone_3 = pytz.timezone('US/Pacific')
+                         #^^^^^^^^^^^^^^^^^^^^^^^^^^^> 2 {{The pytz.timezone is created here.}}
+    dt10 = datetime.datetime(2022, 1, 1, tzinfo=some_timezone_3)  # Noncompliant {{Don't pass a "pytz.timezone" to the "datetime.datetime" constructor.}}
+                                        #^^^^^^^^^^^^^^^^^^^^^^
+
+def with_valid_timezone_alternative():
+    import datetime
+    class ValidTimeZone(datetime.tzinfo):
+        ...
+
+    cond = True
+    if cond:
+        some_timezone = ValidTimeZone()
+    else:
+        some_timezone = pytz.timezone('US/Eastern')
+                       #^^^^^^^^^^^^^^^^^^^^^^^^^^^> 1 {{The pytz.timezone is created here.}}
+    if cond: # This is technically wrong because we will never use the pytz.timezone
+        dt11 = datetime.datetime(2022, 1, 1, tzinfo=some_timezone) # Noncompliant {{Don't pass a "pytz.timezone" to the "datetime.datetime" constructor.}}
+                                            #^^^^^^^^^^^^^^^^^^^^
+
+    another_timezone = ValidTimeZone()
+    aliased_timezone = another_timezone
+    another_timezone = pytz.timezone('US/Eastern')
+    dt12 = datetime.datetime(2022, 1, 1, tzinfo=aliased_timezone)
```

Comment:
I'm not sure I understand what you're aiming for with this test.
Are you missing another call to `datetime` that takes `another_timezone` as an argument?

Reply:
This test was mostly to make sure that the ReachingDefinitionsAnalysis was returning the correct symbol, even if the value is overwritten.